packages/devtools_app/lib/src/memory/flutter/memory_controller.dart

```diff
@@ -83,22 +134,56 @@ class MemoryController extends DisposableController
 
     // The memory source has changed, clear all plotted values.
     memoryTimeline.chartData.reset();
+    memoryTimeline.engineChartData.reset();
   }
 
-  void processData([bool reloadAllData = false]) {
-    final args = offline
-        ? memoryTimeline.processMemoryLogFileData()
-        : memoryTimeline.processLiveData(reloadAllData);
+  void recomputeOfflineData() {
+    final args = memoryTimeline.recomputeOfflineData(pruneIntervalDurationInMs);
+    processDataset(args);
+  }
+
+  void recomputeData() {
+    final args = memoryTimeline.recomputeLiveData(pruneIntervalDurationInMs);
+    processDataset(args);
+    // TODO(terry): need to recomputeOffline data?
+  }
 
+  void processDataset(args) {
+    // Add entries of entries plotted in the chart.  Entries plotted
+    // may not match data collected (based on display interval).
     for (var arg in args) {
       memoryTimeline.chartData.addTraceEntries(
         capacityValue: arg[MemoryTimeline.capcityValueKey],
         usedValue: arg[MemoryTimeline.usedValueKey],
         externalValue: arg[MemoryTimeline.externalValueKey],
+        minutesToDisplay: pruneIntervalDurationInMs,
       );
+      memoryTimeline.engineChartData.addTraceEntries(
+        javaValue: arg[MemoryTimeline.javaHeapValueKey],
+        nativeValue: arg[MemoryTimeline.nativeHeapValueKey],
+        codeValue: arg[MemoryTimeline.codeValueKey],
+        stackValue: arg[MemoryTimeline.stackValueKey],
+        graphicsValue: arg[MemoryTimeline.graphicsValueKey],
+        otherValue: arg[MemoryTimeline.otherValueKey],
+        systemValue: arg[MemoryTimeline.systemValueKey],
+        totalValue: arg[MemoryTimeline.totalValueKey],
+        minutesToDisplay: pruneIntervalDurationInMs,
+      );
+
+      if (memoryTimeline.chartData.pruned) {
+        memoryTimeline.startingIndex++;
+      }
     }
   }
 
+  void processData([bool reloadAllData = false]) {
+    final args = offline
+        ? memoryTimeline.processMemoryLogFileData()
+        : memoryTimeline.processLiveData(reloadAllData);
```

Comment:
these method names don't sound like they will return args. Perhaps rename methods or variable to make that more clear.

Reply:
Changed process to fetch.